### ISSUE TITLE
Steam Flatpak on Fedora + KDE + Wayland starts in unusable state

### ISSUE THREAD

**Issue by ac10n** [ValveSoftware/steam-for-linux#10912]
#### Your system information

* Steam client version (build number or date): Flatpak 1.0.0.79
* Distribution (e.g. Ubuntu):  Fedora 40
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes
* Steam Logs: ~/.steam does not exist
* GPU: Nvidia

Fedora 40 chose to default to not including X11. Now I'm using Wayland. Steam starts like the attached screenshot. I've been using Steam with previous versions of Fedora just fine.

![Screenshot_20240520_052420](https://github.com/ValveSoftware/steam-for-linux/assets/5188263/1a2d300e-38a5-4a7b-841b-c3835b658802)

Also, if I use FlatSeal to disable X11, Steam fails to start with this error: 

```
Check your DISPLAY environment variable and make sure that you have enabled X.
If you are running remotely, make sure that you have a remote connection which will allow an X connection.

For more information visit https://support.steampowered.com/kb_article.php?ref=4050-WOJB-0608
```

#### Steps for reproducing this issue:

1.  Install Steam using Flatpak
4. Start Steam

### So the questions are:

1. Is there an alternative way to install Steam under Fedora other than the Flatpak (From the official website, clicking download gives a `.deb` file which is useful for Debian-based distros
2. Is there a way to fix the Flatpak distribution to work properly?
3. Anything else that can help?

**Comment by ac10n** [comment]
OK. I found out that I can install Steam using `sudo dnf install steam` which does not install flatak. But the issue is still the same. Running `xlsclients` I can see the `steamwebhelper` and `steam` are using xwayland and not running in Wayland Natively.

On my computer, any software running on xwayland has some form of problem. I've managed to run anything I care about natively under Wayland except Steam and Spotify.
